torch/csrc/distributed/rpc/init.cpp

```diff
@@ -448,8 +448,8 @@ If the future completes with an error, an exception is thrown.
       "_invoke_remote_torchscript",
       [](const std::string& dstWorkerName,
          const std::string& qualifiedNameStr,
-         const py::args& args,
-         const py::kwargs& kwargs) {
+         py::args args,
```

Comment:
Is it guaranteed behavior that pybind won't release GIL when passing args?

Reply:
Yes, I think pybind holds GIL when preparing `py::args`, and it also hold GIL when destructing `py::args` (which de-refs the contained `py::object`s).

Since if we write,

```cpp
m.def("foo", foo, py::call_guard<gil_scoped_release>());
```

It's equivalent to this, (according to pybind doc, https://pybind11.readthedocs.io/en/stable/reference.html#_CPPv410call_guard),
where GIL is released only within the local scope of user function.
```
m.def("foo", [](args...) {
    gil_scopede_release scope_guard;
    return foo(args...); // Notice after returning, `scope_guard` is destructed before `args` being desturcted, so GIL is re-acquired before destructing `args`.
});
```